api/src/test/java/org/openmrs/api/EncounterServiceTest.java

```diff
@@ -2959,4 +2959,30 @@ public void shouldSaveMultipleOrderGroupsIfDifferentOrdersHaveDifferentOrderGrou
 		
 		assertEquals("Two New Order Groups Get Saved", 2, orderGroups.size());
 	}
+
+	/**
+	 * @see EncounterService#getEncountersByPatient(String,boolean)
+	 */
+	@Test
+	@Verifies(value = "get all unvoided encounters for the given substring of patient names", method = "getEncountersByPatient(String,boolean)")
+	public void getEncountersByPatient_shouldGetAllUnvoidedEncountersForTheGivenSubstringOfPatientNames() throws Exception {
+		EncounterService encounterService = Context.getEncounterService();
+		
+		List<Encounter> encounters = encounterService.getEncountersByPatient("Joh", false);
+		assertEquals(3, encounters.size());
+	}
+	
+	/**
+	 * @see EncounterService#getEncountersByPatient(String,boolean)
+	 */
+	@Test
+	@Verifies(value = "not get encounters for the given substring of patient identifiers", method = "getEncountersByPatient(String,boolean)")
+	public void getEncountersByPatient_shouldNotGetEncountersForTheGivenSubstringOfPatientIdentifiers() throws Exception {
+		EncounterService encounterService = Context.getEncounterService();
+		
```

Comment:
Would be nice to have an assertion here that ensures there is actually at least one patient with one non voided identifier that matches the search text. Otherwise this could be passing just because there is no match in the test dataset